content/installation/python/PythonInstall.md

```diff
@@ -80,37 +134,71 @@ if __name__ == '__main__':
     app.run(...)
 ```
 
+See the [Flask
+documentation](https://flask.palletsprojects.com/en/1.1.x/quickstart/#hooking-in-wsgi-middleware)
+for more details on using WSGI middleware.
+
 #### Pyramid
 
+In Pyramid, middlewares are called "tweens". See the example below for adding
+the Contrast middleware to a Pyramid app:
+
 ``` python
 from pyramid.config import Configurator
 config = Configurator()
 
 config.add_tween('contrast.agent.middlewares.pyramid_middleware.PyramidMiddleware')
 ```
 
+See the [Pyramid
+documentation](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/environment.html#explicit-tween-configuration)
+for additional details on tween configuration.
+
 #### WSGI
 
+WSGI middleware is implemented as a class that wraps the application instance.
+(This is the basis for both Flask and Pylons middleware as well.)
+
 ``` python
+# This line imports the Contrast middleware class from the package
 from contrast.agent.middlewares.wsgi_middleware import WSGIMiddleware as ContrastMiddleware
 
 # other app code
 
 app = get_wsgi_application()
 
+# This line wraps the application instance with the Contrast middleware
 app = ContrastMiddleware(app)
 ```
 
+See the [WSGI spec](https://www.python.org/dev/peps/pep-0333/#middleware-components-that-play-both-sides)
+for additional details on WSGI middleware.
+
 #### Pylons
 
+Like Flask, Pylons middleware is WSGI middleware, which means the
+middleware class is used to wrap the application instance:
+
 ``` python
 from pylons.wsgiapp import PylonsApp
+# This line imports the Contrast middleware class from the package
 from contrast.agent.middlewares.wsgi_middleware import WSGIMiddleware as ContrastMiddleware
  
 app = PylonsApp()
+# This line wraps the application instance with the Contrast middleware
 app = ContrastMiddleware(app)
 ```
 
 ## Next Steps
 
-Once the installation process is complete, you can update the agent's [configuration](installation-pythonconfig.html) file. 
+Once the installation process is complete, you can update the agent's [configuration](installation-pythonconfig.html) file.
+
+## Troubleshooting
+
```

Comment:
Troubleshooting has been permanently moved to the Knowledge Base. I'll remove this section and send it to the CS team.

Reply:
Thank you!